src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -53,34 +56,58 @@ public Dissector(String mapping) {
         Collections.sort(saveableFields, new FieldComparator());
     }
 
-    public int dissect(byte[] source, Map<String, Object> map) {
-        int pos = dissectValues(source);
-        ValueResolver resolver = new ValueResolver(source, fieldValueRefMap);
+    public int dissect(byte[] source, Map<String, Object> keyValueMap) {
+        ResolveValue resolveValue = resolve(source);
 
         for (Field field : saveableFields) {
-            field.append(map, resolver);
+            field.append(keyValueMap, resolveValue.resolver);
```

Comment:
> Its just a bag of String keys and Object values

I could figure that out by looking at the type signature. :) But what is their *purpose*?

> Its only useful for testing.

With "testing" you mean your unit tests or for debugging / playing around during development?

In the former case, can you maybe restructure it so it makes generally sense to have this map? In the latter case, I'd remove the additional complexity if possible.